pkg/signatures/engine/engine.go

```diff
@@ -22,18 +22,21 @@ type Config struct {
 	Enabled             bool
 	SignatureBufferSize uint
 	Signatures          []detect.Signature
+	DataSources         []detect.DataSource
 }
 
 // Engine is a signatures-engine that can process events coming from a set of input sources against a set of loaded signatures, and report the signatures' findings
 type Engine struct {
-	signatures      map[detect.Signature]chan protocol.Event
-	signaturesIndex map[detect.SignatureEventSelector][]detect.Signature
-	signaturesMutex sync.RWMutex
-	inputs          EventSources
-	output          chan detect.Finding
-	waitGroup       sync.WaitGroup
-	config          Config
-	stats           metrics.Stats
+	signatures       map[detect.Signature]chan protocol.Event
+	signaturesIndex  map[detect.SignatureEventSelector][]detect.Signature
+	signaturesMutex  sync.RWMutex
+	inputs           EventSources
+	output           chan detect.Finding
+	waitGroup        sync.WaitGroup
+	config           Config
+	stats            metrics.Stats
+	dataSources      map[string]map[string]detect.DataSource
+	dataSourcesMutex sync.RWMutex
 }
```

Comment:
I know we did not have comments or the existing maps of maps (example: I needed to refresh memory going to older code for the singaturesIndex map of arrays). Could we add some comments here (why map[string]map[string], since string is too generic to understand at a first glance).

Reply:
Sure, i'll add for the other while i'm at it too.

Reply:
TODO: check if you want to address this after freeze (freezing tomorrow). This will be merged as is by https://github.com/aquasecurity/tracee/pull/3142